README.md

```diff
@@ -53,98 +53,102 @@ If you need to install Go, follow [the official instructions](https://golang.org
-      AddressSpace: &network.AddressSpace{
-        AddressPrefixes: &[]string{"10.0.0.0/8"},
-      },
-      Subnets: &[]network.Subnet{
-        {
-          Name: to.StringPtr("<subnet1Name>"),
-          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
-            AddressPrefix: to.StringPtr("10.0.0.0/16"),
-          },
-        },
-        {
-          Name: to.StringPtr("<subnet2Name>"),
-          SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
-            AddressPrefix: to.StringPtr("10.1.0.0/16"),
-          },
-        },
-      },
-    },
-  },
-  nil)
+func main() {
+	vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+
+	if err == nil {
+		vnetClient.Authorizer = authorizer
+	}
+
+	vnetClient.CreateOrUpdate(context.Background(),
+		"<resourceGroupName>",
+		"<vnetName>",
+		network.VirtualNetwork{
+			Location: to.StringPtr("<azureRegion>"),
+			VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+				AddressSpace: &network.AddressSpace{
+					AddressPrefixes: &[]string{"10.0.0.0/8"},
+				},
+				Subnets: &[]network.Subnet{
+					{
+						Name: to.StringPtr("<subnet1Name>"),
+						SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+							AddressPrefix: to.StringPtr("10.0.0.0/16"),
+						},
+					},
+					{
+						Name: to.StringPtr("<subnet2Name>"),
+						SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+							AddressPrefix: to.StringPtr("10.1.0.0/16"),
+						},
+					},
+				},
+			},
+		})
+}
 ```
 
 ### Authentication
 
-Most SDK operations require an OAuth token for authentication and authorization.
+Most SDK operations require an OAuth token for authentication and authorization. These are
+made available in the Go SDK For Azure through types implementing the `Authorizer` interface.
 You can get one from Azure Active Directory using the SDK's
-[authentication](https://godoc.org/github.com/Azure/go-autorest/autorest/adal) package
-and an existing service principal (aka Client ID and secret) as shown in the
-following example.
-
-The `OAuthTokenProvider` returned by this sample should be set as the
-authorizer for the resource client, as shown in the [previous section](#use).
-
-Note: To create a new service principal, run
+[authentication](https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth) package. The `Authorizer` returned  should
+be set as the authorizer for the resource client, as shown in the [previous section](#use).
+
+You can get an authorizer in the following ways:
+1. From the **Environment**:
+  - Use `auth.auth.NewAuthorizerFromEnvironment()`. This call will try to get an authorizer based on the environment
+variables in the following order: Client Credentials, Client Certificate, Username Pasword, MSI.
+  - The environment variables available are:
+    - `AZURE_TENANT_ID`: Specifies the Tenant to which to authenticate. Not required for MSI auth.
```

Comment:
>Not required for MSI auth. [](start = 72, length = 26)

Instead of specifying MSI does not need this here, I would just specify they are needed for Client cred auth

Reply:
you mean just to enumerate the ones where it is needed ?

Reply:
Yes

Reply:
About MSI, it can be specified it does not need env vars, for example

Reply:
Well it can take the resource as a env var. let me see if I can regroup them